src/stream/stream/mod.rs

```diff
@@ -313,6 +315,41 @@ extension_trait! {
             TakeWhile::new(self, predicate)
         }
 
+        #[doc = r#"
+            Limit the amount of items yielded per timeslice in a stream.
+
+            # Examples
+            ```ignore
+            # fn main() { async_std::task::block_on(async {
+            #
+            use async_std::stream;
+            use std::time::Duration;
+
+            // emit value every 1 second
+            let s = stream::interval(Duration::from_nanos(1000000)).enumerate();
```

Comment:
Perhaps if we use `stream::interval(Duration::from_millis(5)).take(5))` or something similar we could convert this into an automated test. It'd be great if we could remove the `no_run` tag at the top so it tests the functionality.